lib/Transforms/SimplifyParsers.cpp

```diff
@@ -0,0 +1,115 @@
+#include <llvm/Support/ErrorHandling.h>
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "p4mlir/Transforms/Passes.h"
+
+#define DEBUG_TYPE "p4hir-simplify-parsers"
+
+using namespace mlir;
+
+namespace P4::P4MLIR {
+#define GEN_PASS_DEF_SIMPLIFYPARSERS
+#include "p4mlir/Transforms/Passes.cpp.inc"
+
+namespace {
+struct SimplifyParsers : public impl::SimplifyParsersBase<SimplifyParsers> {
+    void runOnOperation() override;
+
+ private:
+    /// Finds all states reachable from the 'start' state.
+    llvm::DenseSet<P4HIR::ParserStateOp> findReachableStates(P4HIR::ParserOp &parser,
+                                                             mlir::SymbolTable &symbolTable);
```

Comment:
`symbolTable` is unused now. Pass `ParserOp` by value here (it is just a wrapper over pointer).